hera_cal/lst_stack/config.py

```diff
@@ -665,6 +668,84 @@ def _read(cls, group: h5py.Group):
         return cls(**dct)
 
 
+class LSTBinConfiguratorSingleBaseline():
+    '''This is a much simpler, less fully featured version of LSTBinConfigurator,
+    designed for LST-binning single baseline (generally 4-pol) files together
+    across nights. Its main goal is '''
+
+    def __init__(self, datadir: str | Path, nights: list[str], fileglob: str):
+        '''Constuctors object from a datadir, nights, and fileglob.
+
+        Parameters
+        ----------
+        datadir : str or Path
+            The directory where the analysis takes place. When combined with fileglob,
+            this should give complete paths to single-baseline, full-night files.
+        nights : list of str
+            A list of JD nights to include. Each night should be a string that can be
+            used to replace the "{night}" in fileglob.
+        fileglob : str
+            A string glob pattern to search for files. This string should contain at least
+            one instance of "{night}" to be replaced by JDs in ``nights``. Likewise, the
+            string should also contain an instance of "{baseline}" which can be replaced
+            by the baseline name (e.g. "0_1") to find the file for that night and baseline.
+        '''
+        self.datadir = datadir
+        self.nights = nights
+        self.fileglob = fileglob
+        self.bl_to_file_map = self.build_bl_to_file_map()
+
+    @classmethod
+    def from_toml(cls, toml_file: str | Path) -> LSTBinConfiguratorSingleBaseline:
+        '''Build a LSTBinConfiguratorSingleBaseline from a toml file
+
+        Parameters
+        ----------
+        toml_file : str or Path
+            The path to a toml file has makeflow_type = "lstbin_single_baseline".
+            The toml file should have a section called "FILE_CFG" with a sub-section
+            called "datafiles" that contains "datadir", "nights", and "fileglob" keys.
+        '''
+        dct = toml.load(Path(toml_file))
+        if dct['Options']['makeflow_type'] != 'lstbin_single_baseline':
+            raise ValueError(f"Expected 'makeflow_type' to be 'lstbin_single_baseline', "
+                             f"but got {dct['Options']['makeflow_type']}.")
+        datafiles_cfg = dct['FILE_CFG']['datafiles']
+        return cls(datafiles_cfg['datadir'], datafiles_cfg['nights'], datafiles_cfg['fileglob'])
```

Comment:
Let's use keywords instead of position args here

Reply:
done